src/DnsHelper.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Yiisoft\NetworkUtilities;
+
+class DnsHelper
+{
+    /**
+     * @param string $hostname hostname without dot at end
+     * @link https://bugs.php.net/bug.php?id=78008
+     */
+    public static function checkMx(string $hostname): bool
+    {
+        $hostname .= '.';
```

Comment:
I'd make it fail-safe

```suggestion
        $hostname = rtrim($hostname, '.') . '.';
```